src/connectedmachine/azext_connectedmachine/generated/_params.py

```diff
@@ -45,66 +50,174 @@ def load_arguments(self, _):
     with self.argument_context('connectedmachine extension show') as c:
         c.argument('resource_group_name', resource_group_name_type)
         c.argument('machine_name', type=str, help='The name of the machine containing the extension.', id_part='name')
-        c.argument('name', options_list=['-n', '--extension-name', '--name'], type=str, help='The name of the machine '
-                   'extension.', id_part='child_name_1')
+        c.argument('name', type=str, help='The name of the machine extension.', id_part='child_name_1')
 
     with self.argument_context('connectedmachine extension create') as c:
         c.argument('resource_group_name', resource_group_name_type)
         c.argument('machine_name', type=str, help='The name of the machine where the extension should be created or '
                    'updated.')
-        c.argument('name', options_list=['-n', '--extension-name', '--name'], type=str, help='The name of the machine '
-                   'extension.')
+        c.argument('name', type=str, help='The name of the machine extension.')
         c.argument('tags', tags_type)
-        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), required=False,
```

Comment:
What will happen if the user does not pass in `location`?

Reply:
location is not a required param for creating an extension.